e2e2/test/cases/nvidia/main_test.go

```diff
@@ -129,15 +139,162 @@ func checkNodeTypes(ctx context.Context, config *envconf.Config) (context.Contex
 	return ctx, nil
 }
 
+func createMetricManager(ctx context.Context, config *envconf.Config) (context.Context, error) {
+	log.Printf("AMP url is set to %s", *ampMetricUrl)
+	metadataLabel, err := getMetadataLabel(ctx, config)
+	if err != nil {
+		return ctx, err
+	}
+	metricManager = metric.NewMetricManager(metadataLabel, awsCfg, *ampMetricUrl, *ampMetricRoleArn)
+
+	return ctx, nil
+}
+
+func getMetadataLabel(ctx context.Context, cfg *envconf.Config) (map[string]string, error) {
+	restConfig := cfg.Client().RESTConfig()
+	// Initialize Kubernetes client from the provided REST config
+	clientset, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create Kubernetes client: %v", err)
+	}
+
+	// List nodes in the cluster
+	nodes, err := clientset.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to list nodes: %v", err)
+	}
+
+	// Ensure at least one node is available
+	if len(nodes.Items) == 0 {
+		return nil, fmt.Errorf("no nodes found in the cluster")
+	}
+
+	// Get instance type and metadata from the first node
```

Comment:
the test case shouldn't really assume that all the nodes in the cluster are the same across all these dimensions; can you pass in the dimensions with your sample, instead of fetching them ahead of time? Then you'd be able to pass dimensions that you know match the sample

Reply:
Updated in the latest revision